.github/workflows/testAndPublish.yml

```diff
@@ -0,0 +1,314 @@
+name: Run tests and publish
+
+on:
+  push:
+    branches:
+    - master
+    - beta
+    - rc
+    - 'try-**'
+    tags:
+    - 'release-**'
+
+  pull_request:
+    branches:
+    - master
+    - beta
+    - rc
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  START_BUILD_NUMBER: 50000
+  pullRequestNumber: ${{ github.event_name == 'pull_request' && github.event.number || 0 }}
+  scons_publisher: NV Access
+  # Comment out any of the feature_* variables to disable the respective build feature.
+  # They are checked for existence of content, not specific value.
+  # feature_buildSymbols: configured
+  # feature_uploadSymbolsToMozilla: configured
+  # feature_buildAppx: configured
+  # feature_crowdinSync: configured
+  # feature_signing: configured
+
+jobs:
+  buildNVDA:
+    name: Build NVDA
+    runs-on: windows-latest
+    steps:
+    - name: Checkout NVDA
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+    - name: Install dependencies
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+        architecture: x86
+    - name: Set version variables
+      run: ci/scripts/setBuildVersionVars.ps1
+    - name: Set scons args
+      run: ci/scripts/setSconsArgs.ps1
+    - name: Prepare source code
```

Comment:
I probably missed why we no longer cache the venv as we did before. I think it might make sense to do so because it can really speed up builds.

Reply:
The cache in the original design wasn't being utilized properly, it was caching, but not using the cache.  I'm not particularly convinced we can cache `.venv` on requirements alone, which was the key, for example what if we change our virtual env scripts?
With caching the entire build, we would be double caching .venv. We have a limit of 10GB/repo, which means about 30 (potentially parrallel) builds before we start losing caches mid build.

Reply:
Another particular concern with `.venv` caching would be changes to build env such as the python interpreter. I don't think there's a particularly clear way of predetermining the expected cache based on variables available .venv generation